pkg/service/contract/service.go

```diff
@@ -9,8 +9,8 @@ import (
 
 // Service defines the contract service interface
 type Service interface {
-	Generate(ctx context.Context) error
-	Download(ctx context.Context) error
+	Generate(ctx context.Context, checkConfig bool) error
```

Comment:
Rather than passing as an argument, Can't you just pass the information to the constructor of the contract and decide whether it should be true or not in the validateFlags function?